semeio/workflows/localisation/local_script_lib.py

```diff
@@ -286,22 +539,86 @@ def add_ministeps(
+                    data_size2 = field_config.get_data_size()
+                    assert data_size == data_size2
+                    param_for_field = None
+                    if corr_spec.field_scale.method == "gaussian_decay":
+                        ref_pos = corr_spec.ref_point
+                        main_range = corr_spec.field_scale.main_range
+                        perp_range = corr_spec.field_scale.perp_range
+                        azimuth = corr_spec.field_scale.azimuth
+                        check_if_ref_point_in_grid(ref_pos, grid_for_field)
+                        apply_gaussian_decay(
+                            row_scaling,
+                            data_size,
+                            grid_for_field,
+                            ref_pos,
+                            main_range,
+                            perp_range,
+                            azimuth,
+                        )
+
+                    elif corr_spec.field_scale.method == "exponential_decay":
+                        ref_pos = corr_spec.ref_point
+                        main_range = corr_spec.field_scale.main_range
+                        perp_range = corr_spec.field_scale.perp_range
+                        azimuth = corr_spec.field_scale.azimuth
+                        check_if_ref_point_in_grid(ref_pos, grid_for_field)
+                        apply_exponential_decay(
+                            row_scaling,
+                            data_size,
+                            grid_for_field,
+                            ref_pos,
+                            main_range,
+                            perp_range,
+                            azimuth,
+                        )
+
+                    elif corr_spec.field_scale.method == "from_file":
+                        apply_from_file(
+                            row_scaling,
+                            data_size,
+                            grid_for_field,
+                            corr_spec.field_scale.filename,
+                            corr_spec.field_scale.param_name,
+                            user_config.log_level,
+                        )
+
+                    elif corr_spec.field_scale.method == "segment":
+                        param_for_field = apply_segment(
+                            row_scaling,
+                            data_size,
+                            grid_for_field,
+                            corr_spec.field_scale.segment_filename,
+                            corr_spec.field_scale.param_name,
+                            corr_spec.field_scale.active_segments,
+                            corr_spec.field_scale.scalingfactors,
+                            corr_spec.field_scale.smooth_ranges,
+                            corr_spec.name,
+                            user_config.log_level,
+                        )
+                    else:
+                        logging.warning(
```

Comment:
This should be an error.

Reply:
Should I use raise ValueError(...) or logging.error(...)  ?